clippy_lints/src/attrs.rs

```diff
@@ -946,6 +954,19 @@ fn check_nested_misused_cfg(cx: &EarlyContext<'_>, items: &[NestedMetaItem]) {
             }
             if let MetaItemKind::List(list) = &meta.kind {
                 check_nested_misused_cfg(cx, list);
+            // If this is not a list, then we check for `cfg(test)`.
+            } else if let Some(ident) = meta.ident()
+                && matches!(ident.name.as_str(), "tests" | "Test")
```

Comment:
Instead of deny-listing `tests` and `Test`, would it be possible to allow-list all configs passed in by `--cfg` plus the built-in ones (`test`, `debug_assertions`, …)?

Reply:
I'll check for the full list then.

Reply:
So back to this, I'm not sure it's a good approach because you might use a valid `cfg` that is not passed to the compiler (so it's `cfg`ed-out) and in this case it would be considered as an error since this `cfg` isn't known to clippy. So false-positive.

However on the other hand, it's already possible to have this issue currently, so not sure if we want to dive into this any further.

I think a next iteration on this would be to eventually check for all existing ones (the builtin ones and the ones provided to rustc) but we need to check pros and cons.

Does that sound acceptable?

Reply:
You are right; catching all `cfg` errors is definitely a different and much larger issue

Reply:
Isn't that https://doc.rust-lang.org/beta/unstable-book/compiler-flags/check-cfg.html?

Reply:
Yes it is. I discovered it afterwards and I linked to the tracking issue in my last comment of this PR.